### DISCUSSION TITLE
flutter with fvm

### DISCUSSION THREAD

**Discussion by mostafejur21** [neovim/neovim#27471] [Q&A] [answered]
hello. how can i setup fvm in neovim lsp? i try flutter-tools. but cannot setup the fvm. is there any way to do that?
i'm using lazy.nvim.

**Comment by olrtg** [comment]
This is my config: https://github.com/olrtg/dotfiles/blob/c294a72012f527ffb0866541ee26128248803738/lvim/lua/user/languages/flutter.lua. I'm using LunarVim but maybe you can extrapolate from that to your config.

**Reply by mostafejur21** [reply]
how do you toggle different fvm version?

**Reply by olrtg** [reply, marked as answer]
You can change it via telescope https://github.com/akinsho/flutter-tools.nvim?tab=readme-ov-file#fvm. Make sure to enable that integration first https://github.com/akinsho/flutter-tools.nvim?tab=readme-ov-file#telescope-integration

**Reply by mostafejur21** [reply]
thank you so much. it works fine.

**Comment by mostafejur21** [comment]
i got this error every time i open neovim.
"Error detected while processing C:\Users\ysose\AppData\Local\nvim\init.lua:
E5113: Error while calling lua chunk:
C:\Users\ysose\AppData\Local\nvim\init.lua:799: attempt to concatenate a
nil value
stack traceback:
        C:\Users\ysose\AppData\Local\nvim\init.lua:799: in main chunk"
although i've enter the current path of my fvm
here is the path that i entered.
"flutter_path = os.getenv("HOME") .. "/fvm/default/bin/","
and i got that error every time.

**Reply by olrtg** [reply]
From what the error says I suppose that `os.getenv("HOME")` is `nil` and you're trying to concatenate that to the other string. From what I see too is that you're using Windows but I don't know if the `$HOME` variable exists on Windows. But I'm sure that my config works on Linux and macOS.

**Comment by mostafejur21** [comment]
I switch to linux today. And try to run same config. But my linux config
also give me error message if i add the path line. But if i remove the path
line, it does not gives me error.

**Reply by olrtg** [reply]
Well, can you share that part of your config? or may share the link if it's in a public repo?

**Comment by mostafejur21** [comment]
sure. Here is my config link. i just try to things work in kickstart nvim.
kindly give me some idea for setup my configuration
https://github.com/mostafejur21/kickstart.nvim/blob/master/init.lua#L691

**Reply by olrtg** [reply]
Try `flutter_path = os.getenv("HOME") .. "/fvm/default/bin/flutter"` (note the `/flutter`)

**Comment by mostafejur21** [comment]
I can run the Telescope flutter fvm command and select the fvm version. But
it does not effect anything. And the FlutterEmulators and FlutterDevices
command does not work.

**Reply by olrtg** [reply]
Well, that's as far as my knowledge goes 😅. Try raising an issue in the flutter-tools.nvim repo or in the discussions section if they have one.

**Comment by mostafejur21** [comment]
Thank you for your time and response.
